webapp/public/logic.py

```diff
@@ -76,17 +76,28 @@ def convert_navigation_url(url, link):
     url_parsed = urlparse(link)
     host_url = (
         "{base_url}"
-        "?q={q}&limit={limit}&offset={offset}"
+        "?q={q}&category={category}&limit={limit}&offset={offset}"
     )
 
     url_queries = parse_qs(url_parsed.query)
-    q = url_queries['q'][0]
+
+    if 'q' in url_queries:
+        q = url_queries['q'][0]
+    else:
+        q = ''
+
+    if 'section' in url_queries:
+        category = url_queries['section'][0]
+    else:
+        category = ''
+
     size = int(url_queries['size'][0])
     page = int(url_queries['page'][0])
 
     return host_url.format(
         base_url=url,
         q=q,
+        category=category,
```

Comment:
I think you should add the category after the url is built, otherwise it makes an ugly url if you are not searching for particular categories: http://127.0.0.1:5000/search?category=&q=toto